lib/engine/game/g_1854/depot.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require_relative '../../depot'
+
+module Engine
+  module Game
+    module G1854
+      class Depot < Engine::Depot
+        def depot_trains(clear: false)
+          d_trains = super
+          d_trains.reject! { |t| t.name == '3+' } if @game.depot.upcoming.any? { |t| t.name == '2+' }
+          d_trains.reject! { |t| t.name == '2+' } if @game.depot.upcoming.any? { |t| t.name == '1+' }
+          d_trains
```

Comment:
You should maintain the state of the Depot `@depot_trains` variable here. Maybe something like:
```suggestion
          @depot_trains = nil if clear
          if @depot_trains.nil?
                    @depot_trains = super
                    @depot_trains.reject! { |t| t.name == '3+' } if @game.depot.upcoming.any? { |t| t.name == '2+' }
                    @depot_trains.reject! { |t| t.name == '2+' } if @game.depot.upcoming.any? { |t| t.name == '1+' }
          end
          @depot_trains
```

Reply:
Except that's really quite ugly. It might be cleaner to refactor the base method `Engine::Depot.depot_trains` to look something like this:
```
    def depot_trains(clear: false)
      @depot_trains = nil if clear
      @depot_trains ||= available_upcoming_trains + @discarded.uniq(&:name)
    end

    def available_upcoming trains
       [@upcoming.first,
        *@upcoming.select { |t| @game.phase.available?(t.available_on) },
      ].compact.uniq(&:name)
    end
```
You could then override the `available_upcoming_trains` method in your Depot subclass.